src/index.js

```diff
@@ -1,47 +1,97 @@
-            <img src=${Refresh} alt="refresh" class="todo-hold">
-    </li>
-    <li class="todo-item todo-add">
-        <input type="text" placeholder="Add to your list...">
-    </li>`;
-  todosArr.forEach((todo) => {
-    todoItems = `${todoItems} <li class="todo-item todos">
-        <div class="todo-wrap">
-            <input type="checkbox">
-            <p class="todo-text"> ${todo.description}</p>
-        </div>
-        <img src="${Icon}" alt="move" class="todo-hold">
-        </li>`;
-  });
+  let todoItems = '';
+  refresImg.src = Refresh;
+  btnAddTodo.src = Add;
+  const storageTasks = JSON.parse(localStorage.getItem('todos')) || [];
+  if (tasksArr !== null) {
+    storageTasks.forEach((todo, index) => {
+      todoItems = `${todoItems} <li class="todo-item todos">
+          <div class="todo-wrap">
+              <input type="checkbox">
+              <input id="todo-${index + 1}" type="text" class="todo-text" value="${todo.description}">
+          </div>
+          <img src="${Edit}" id="edit-${index + 1}" alt="edit" class="edit-btn todo-hold">
+          <img src="${Remove}" id="${index + 1}" alt="edit" class="remove-btn todo-hold">
+          <img src="${Icon}" alt="move" class="todo-hold">
+          </li>`;
+    });
+  }
   todoItems = `${todoItems} <li class="todo-item todo-button">
-    <button>Clear all completed</button>
-</li>`;
+      <button id="btn-clear">Clear all completed</button>
+  </li>`;
   todoListUl.innerHTML = todoItems;
 };
 
-window.onload = populateTodo();
\ No newline at end of file
+const removeTodo = (event) => {
+  const btnId = Number(event.target.id);
+  tasksArr.splice(btnId - 1, 1);
+  tasksArr.forEach((todo, index) => {
+    todo.index = index + 1;
+  });
+  localStorage.setItem('todos', JSON.stringify(tasksArr));
+  populateTodo();
+  addDeleteEvents(removeTodo);
+  addEditEvents(editTodo);
+};
+
+editTodo = (event) => {
+  const todoTextInput = document.querySelectorAll('.todo-text');
+  const eventItemId = event.target.id;
+  let newDescription = '';
+  const matches = eventItemId.replace(/^\D+/g, '');
+  const editId = Number(matches);
+  const todoTextInputArr = Array.from(todoTextInput);
+  todoTextInputArr.forEach((input) => {
+    const inputId = input.id;
+    const extractId = Number(inputId.replace(/^\D+/g, ''));
+    if (editId === extractId) {
+      newDescription = input.value;
+    }
+  });
+  tasksArr.forEach((todo, index) => {
+    if (editId - 1 === index) {
+      todo.description = newDescription;
+    }
+  });
+  localStorage.setItem('todos', JSON.stringify(tasksArr));
+  populateTodo();
+  addDeleteEvents(removeTodo);
+  addEditEvents(editTodo);
+};
+
+const addTodo = () => {
+  const todotext = todoAddInput.value;
+  const todo = new Todo(todotext, false, tasksArr.length + 1);
+
+  tasksArr.push(todo);
+  todoAddInput.value = '';
+  localStorage.setItem('todos', JSON.stringify(tasksArr));
+  populateTodo();
+  addDeleteEvents(removeTodo);
+  addEditEvents(editTodo);
+};
```

Comment:
- [Optional] You did well implementing the requested changes, One thing I notice is when I hit the `enter` key on my keyboard your page reloads. it would be better to have an event listener for the `enter` key to add a task than reload a page. This is for a better user experience.